### DISCUSSION TITLE
Looking for guidance on the compiler explorer

### DISCUSSION THREAD

**Discussion by kxing28** [NUS-CG3207/labs#28] [Lab 4] [answered]
I'm looking for resources on compiling C code into RISC-V. I can't find the right settings on the [Compiler Explorer](https://godbolt.org/). For example, when we compile
```
#include <stdio.h>
int main() {
   // printf() displays the string inside quotation
   printf("Hello, World!");
   return 0;
}
```
with RISC-V 32 bits gcc 12.2.0, we get

```
.LC0:
        .string "Hello, World!"
main:
        addi    sp,sp,-16
        sw      ra,12(sp)
        sw      s0,8(sp)
        addi    s0,sp,16
        lui     a5,%hi(.LC0)
        addi    a0,a5,%lo(.LC0)
        call    printf
        li      a5,0
        mv      a0,a5
        lw      ra,12(sp)
        lw      s0,8(sp)
        addi    sp,sp,16
        jr      ra
```

Pasting this directly into RARS requires we add `.data` before `.LC0` and `.text` before `main`. But there's also
```
call printf
```
which comes from `stdio.h`. We use UART to write to console, so is there some conversion we can do to translate `printf` to UART? 

Also, what compiler flags do we need to have library functions compiled into the assembly code? My partner and I are considering using `<termios>` to draw pictures to Console via UART.

Finally, gcc can output pseudoinstructions/syntax that RARS doesn't interpret. For example,
```
sw      a4,%lo(_UPDATE_FREQUENCY)(a5)
```

is invalid in RARS. How can we configure the compiler to produce the right instructions?

**Comment by r-panicker** [comment, marked as answer]
I guess most of the issues mentioned above are already covered in the manual - https://nus-cg3207.github.io/lab-manuals/lab4/Using_Compiled_Code.html
